format.md

```diff
@@ -0,0 +1,133 @@
+# About this document
+
+This document attempts to document the format currently implemented by binjs-fbssdc, also known as "context 0.1".
+
+# Global structure
+
+```
+Stream ::= MagicHeader BrotliBody
+```
+
+# Magic Header
+
+The magic header serves to identify the format.
+
+```
+MagicHeader ::= "\x89BJS\r\n\0\n" FormatVersion
+FormatVersion ::= 0b00000010
+```
+
+# Brotli content
+
+With the exception of the header, the entire file is brotli-compressed.
+
+```
+BrotliBody ::= Brotli(Body)
+Body ::= Prelude AST
+```
+
+Where `Brotli(...)` represents data that may be uncompressed by the
+`brotli` command-line tool or any compatible library.
+
+# Prelude
+
+The prelude defines a dictionary of strings and a dictionary of probabilities. The order of both is meaningful.
+
+```
+Prelude ::= StringPrelude ProbabilityPrelude
+```
+
+## String dictionary
+
+```
+StringPrelude ::= n=NumberOfStrings StringDefinition{n}
+NumberOfStrings ::= varnum
+StringDefinition ::= NonZeroByte* ZeroByte
+NonZeroByte ::= 0x01-0xFF
+ZeroByte ::= 0x00
+```
+
+Strings are utf-8 encoded, then we replace any embedded `0x00 0x01` with `0x01 0x01` (FIXME: Why does this work?)
+
+
+
+## Probability dictionaries
+
```

Comment:
how about adding something like the following?
(just suggestion for later changes, not that they should be added in this PR)

 * the dictionary defines the possible AST, and code for each possibility to encode the actual tree in tree part of the file
 * the possible AST is the subset of the valid ECMAScript AST
 * If there's only one possibility at some node, it's fully defined in the dictionary
 * if there's 2 or more possibilities at some node, code in tree part defines which possibility to choose at that node
 * primitives values (excluding strings) and the length of lists are all defined in this dictionary, not in tree part

Reply:
I've tried clarifying in the latest version.